### PR TITLE
Add `client.delete_post`

### PR DESCRIPTION
**Pull request by rmeritz**
Deletes a post with force (regardless of if it is published or not).

### CODE REVIEW

**Comment by Mange** [comment]
I want that to be a parameter, actually. It should default to `false`, just like WP does.

I'm also okay with having `force: true` swallow any `NotFoundError`, just like `rm -f` and other UNIX tools.

**Comment by rmeritz** [comment]
> I want that to be a parameter, actually. It should default to false, just like WP does.

I just tried rewritting it with force as an optional argument. When I wrote integration tests for force false and the post as a draft I was not able to delete it. 

I was originally just guessing what force meant, but it seems that I was wrong and that is just a required argument. So I'm going to leave it as is.

**Comment by rmeritz** [comment]
> I just tried rewritting it with force as an optional argument. When I wrote integration tests for force false and the post as a draft I was not able to delete it. 

It turns out I was able to delete it I just wasn't easily able to tell that I could delete it because I was still able to fetch it because if you just delete it without force it moves into the trash can but is still fetchable via the API.

I implemented a force delete option.

File: lib/wpclient/client.rb [hunk 1]
```diff
@@ -113,6 +113,14 @@ def upload_file(filename, mime_type:)
       end
     end
 
+    def delete_post(id, force: false)
+      attrs = {}
+      if force
+        attrs["force"] = true
+      end
+      connection.delete("posts/#{id.to_i}", attrs)
```
Comment (Mange) [inline lib/wpclient/client.rb]: Wouldn't it just work by passing `force: force` to `#delete`?

File: spec/client_spec.rb [hunk 2]
```diff
@@ -233,6 +233,32 @@
...
+
+    it "requests that a client does not forcefully delete a post" do
+      expect(connection).to receive(:delete).with(
+        "posts/1", {}
+      ).and_return true
+
+      expect(client.delete_post(1, force: false)).to eq true
+    end
+
+    it "requests that a client forcefully deletes a post" do
```
Comment (Mange) [inline spec/client_spec.rb]: `it "can forcefully delete a post"`

File: spec/client_spec.rb [hunk 3]
```diff
@@ -233,6 +233,32 @@
     end
   end
 
+  describe "deleting posts" do
+    it "requests that the client delete a post" do
```
Comment (Mange) [inline spec/client_spec.rb]: This example is made redundant by the next example.

File: spec/client_spec.rb [hunk 4]
```diff
@@ -233,6 +233,32 @@
...
+      expect(connection).to receive(:delete).with(
+        "posts/1", {}
+      ).and_return true
+
+      expect(client.delete_post(1)).to eq true
+    end
+
+    it "requests that a client does not forcefully delete a post" do
```
Comment (Mange) [inline spec/client_spec.rb]: `it "deletes post non-forcefully by default"`

**Comment by Mange** [comment]
:+1:

**Pull request merged**